src/client/pydaos/pydaos_shim.c

```diff
@@ -569,16 +572,33 @@ __shim_handle__kv_get(PyObject *self, PyObject *args)
 				if (rc != DER_SUCCESS)
 					D_GOTO(err, 0);
 			} else if (evp->ev_error == -DER_REC2BIG) {
-				/**
-				 * op->size = VAL_SZ;
-				 * D_REALLOC(op->buf, op->buf, op->size);
-				 */
+				char *new_buff;
+
+				op->size *= 2;
```

Comment:
The actual size is supposed to be reported back by the server, we should just use this IMO.

Reply:
I was thinking that would be a much better way of doing it and did check the API to see if that was happening but couldn't see anything and didn't ask.  Will fix.

Reply:
Looks like the actual size is only being reported back on successful read.

Reply:
Let's check this with Mohamad.